From 16d72de3280410b185f33e489727761096a2e6e4 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Wed, 22 Oct 2025 14:34:22 +0300
Subject: [PATCH] decode: fix build on ARMv7 targets

Building NEON code on ARMv7 fails with the following error:

...../src/NextStartCodeNEON.cpp:39:57: error: cannot convert 'uint8x8_t' to 'uint64x1_t'

Add missing vreinterpret call in order to fix that error.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Submitted [https://github.com/KhronosGroup/Vulkan-Video-Samples/pull/109]
---
 vk_video_decoder/libs/NvVideoParser/src/NextStartCodeNEON.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vk_video_decoder/libs/NvVideoParser/src/NextStartCodeNEON.cpp b/vk_video_decoder/libs/NvVideoParser/src/NextStartCodeNEON.cpp
index f1a3381d7122..2c695a924ebd 100644
--- a/vk_video_decoder/libs/NvVideoParser/src/NextStartCodeNEON.cpp
+++ b/vk_video_decoder/libs/NvVideoParser/src/NextStartCodeNEON.cpp
@@ -36,7 +36,7 @@ size_t VulkanVideoDecoder::next_start_code<SIMD_ISA::NEON>(const uint8_t *pdatai
 #if defined (__aarch64__) || defined(_M_ARM64)
                 uint64_t resmask = vmaxvq_u8(vmask);
 #else
-                uint64_t resmask = vget_lane_u64(vmax_u8(vget_low_u8(vmask), vget_high_u8(vmask)), 0);
+                uint64_t resmask = vget_lane_u64(vreinterpret_u64_u8(vmax_u8(vget_low_u8(vmask), vget_high_u8(vmask))), 0);
 #endif
                 if (resmask)
                 {
-- 
2.51.0

